pkg/koordlet/metricsadvisor/collectors/performance/performance_collector_linux.go

```diff
@@ -99,59 +100,50 @@ func (p *performanceCollector) collectContainerCPI() {
 			containerStatusesMap[containerStat] = meta
 		}
 	}
-	// get container CPI collectors for each container
-	collectors := sync.Map{}
+	metricsChan := make(chan metriccache.MetricSample, len(containerStatusesMap)*2)
+	pciMetrics := make([]metriccache.MetricSample, 0)
+	// get container PCI Metric
+	go p.getContainerPCIMetrics(containerStatusesMap, metricsChan)
+
+	for item := range metricsChan {
+		pciMetrics = append(pciMetrics, item)
+	}
+	// save container PCI metric to tsdb
+	p.saveMetric(pciMetrics)
+
+	p.started.Store(true)
+	klog.V(5).Infof("collectContainerCPI for time window %s finished at %s, container num %d",
+		timeWindow, time.Now(), len(containerStatusesMap))
+}
+
+func (p *performanceCollector) getContainerPCIMetrics(containerStatusesMap map[*corev1.ContainerStatus]*statesinformer.PodMeta, metricsChan chan<- metriccache.MetricSample) {
```

Comment:
if metricsChan is for solving the concurrency problem, maybe it is not necessary, just let getContainerPCIMetrics return with a slice is ok, and the slice should append with sync.Mutex.